src/components/button/Button.tsx

```diff
@@ -41,23 +41,28 @@ export const Button: React.FC<ButtonProps> = ({
     prefix,
     `${prefix}--${variant}`,
     iconButton && `${prefix}--icon`,
-    iconRight && `${prefix}--icon-right`,
-    iconLeft && `${prefix}--icon-left`,
-    `${loading ? 'loading' : ''}`,
+    loading && 'loading',
     `${prefix}--${size}`
   );
-  if(variant==='destructive') {
+
+  if(variant === 'destructive') {
     console.warn('The button variant: \'destructive\' will be deprecated.\n Please use: \'primary-destructive\' instead')
   }
+
   return (
     <button
+      role="button"
       className={classes}
       disabled={loading || disabled}
       /* eslint-disable react/button-has-type */
       type={type}
       {...rest}
     >
-      {loading ? <i className="animate-spin tk-icon-loading" /> : iconLeft ? <>{iconLeft}{children}</> : iconRight ? <>{iconRight}{children}</> : children}
+      {loading && <i className="animate-spin tk-icon-loading" />}
+      <span className={classNames(iconRight && `${prefix}--icon-right`,iconLeft && `${prefix}--icon-left`)}
```

Comment:
IMO, classNames should take an object in parameter, isn't it ?
```classNames({ iconRight: `${prefix}--icon-right`, iconLeft: `${prefix}--icon-left` })```
(Se: https://www.npmjs.com/package/classnames)